crates/ra_syntax/src/ast/generated/nodes.rs

```diff
@@ -948,21 +1789,56 @@ impl Visibility {
     pub fn self_token(&self) -> Option<SyntaxToken> { support::token(&self.syntax, T![self]) }
     pub fn crate_token(&self) -> Option<SyntaxToken> { support::token(&self.syntax, T![crate]) }
 }
-
+/// Single identifier.
+/// // TODO: clarify the difference between Name and NameRef
+///
+/// ```
+/// let ❰ foo ❱ = bar;
+/// struct ❰ Baz ❱;
+/// fn ❰ bruh ❱() {}
+/// ```
+///
+/// [Reference](https://doc.rust-lang.org/reference/identifiers.html)
```

Comment:
`Name` is for things that install a new name into the scope, `NameRef` is a usage of a name. Most of the time, this definition/reference distinction can be determined purely syntactically, ie in `fn foo() { foo() }` the first `foo` is `Name`, the second one is `NameRef`. The notable exception are patterns, where in `let x = 92` `x` can be semantically either a name or a name ref, depeding on weather there's an `x` constant in scope. We use `Name` for patterns, and disambiguate semantically (see `NameClass` in ide_db)

Reply:
Thanks, added this citation in
24b27abf9f3a57268d0fb7c9a1aa2ede2980195f